tests/model/ImageTest.php

```diff
@@ -411,6 +411,38 @@ public function testPathPropertiesCachedImage() {
 	}
 
 	/**
+	 * Tests the static function Image::strip_resampled_prefix, to ensure that
+	 * the original filename can be extracted from the path of transformed images, 
+	 * both in current and previous formats 
+	 */
+	public function testStripResampledPrefix() {
+		$orig_image = $this->objFromFixture('Image', 'imageWithoutTitleContainingDots');
+
+		$paths = array();
+
+		// current format (3.3+). Example:
+		// assets/ImageTest/_resampled/ScaleHeightWzIwMF0=/ScaleWidthWzQwMF0=/test.image.with.dots.png;
+		$firstImage = $orig_image->ScaleWidth(200);
+		$secondImage = $firstImage->ScaleHeight(200);
+		$paths[] = $firstImage->Filename;
+		$paths[] = $secondImage->Filename;
+
+		// 3.2 format
+		$paths[] = 'assets/ImageTest/_resampled/ScaleHeightWzIwMF0=-test.image.with.dots.png';
+		//$paths[] = 'assets/ImageTest/_resampled/ScaleHeightWzIwMF0=-ScaleWidthWzQwMF0=-test.image.with.dots.png';
+		// Does not work, because the regex did not support this in 3.2!
+
+		// 3.1 (and earlier) format 
+		$paths[] = 'assets/ImageTest/_resampled/ScaleHeight200-';
+		//$paths[] = 'assets/ImageTest/_resampled/ScaleHeight200-ScaleWidth200-test.image.with.dots.png';
+		// Does not work, because the regex did not support this in 3.1!
+
+		foreach($paths as $path) {
```

Comment:
Generally looped assertions are hard to review, since some test loggers just say "failure on line 441". In future it's preferable to unroll these. :)